### PR TITLE
Show a summary after `meson build`

### PR DESCRIPTION
**Pull request by superm1**
I personally find it really hard to look through all the dependency checking stuff and determine if a plugin is/isn't enabled.
Show a summary at the end of meson build which plugins and features are going to be built based on the equations.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -205,32 +209,39 @@ gmodule = dependency('gmodule-2.0')
 if build_standalone
 gudev = dependency('gudev-1.0', version: '>= 232',
                     required: get_option('gudev').disable_auto_if(host_machine.system() != 'linux'))
+summary({'gudev': gudev}, section: 'project features')
 if gudev.found()
   conf.set('HAVE_GUDEV', '1')
 endif
 bluez = get_option('bluez').disable_auto_if(host_machine.system() != 'linux')
+summary({'bluez': bluez.allowed()}, section: 'project features')
```
Comment (hughsie) [inline meson.build:217]: Isn't the indenting weird?

**Comment by hughsie** [comment]
If it helps, this is what glib does: https://gitlab.gnome.org/GNOME/glib/-/blob/main/meson.build?ref_type=heads#L2534-L2601

**Comment by superm1** [comment]
> If it helps, this is what glib does: https://gitlab.gnome.org/GNOME/glib/-/blob/main/meson.build?ref_type=heads#L2534-L2601

Oh, I think that's a lot cleaner to group it all together in one place.  I'll make some changes.

File: plugins/acpi-dmar/meson.build [hunk 2]
```diff
@@ -1,4 +1,5 @@
 if hsi and (host_cpu == 'x86' or host_cpu == 'x86_64')
+plugins += {meson.current_source_dir().split('/')[-1]: true}
```
Comment (hughsie) [inline plugins/acpi-dmar/meson.build:2]: so i guess we have to do this as the directory name and plugin name are not 1:1 related.

**Review by hughsie — APPROVED** [review]

File: plugins/acpi-dmar/meson.build (hunk 2 above)
Comment (superm1) [reply plugins/acpi-dmar/meson.build:2]: yup

**Pull request merged**